Spatial.gd

```diff
@@ -284,27 +285,34 @@ func reset_minimap_masks():
 		minimap_path.material.set_shader_param("minimap_corner", compass_corner1)
 		minimap_path.material.set_shader_param("minimap_corner2", compass_corner2)
 
-var markerdata = Waypoint.Waypoint.new()
-var marker_file_dir = "user://protobins/"
+var Waypoint_data = Waypoint.Waypoint.new()
+var marker_file_dir = Directory.new()
 var marker_file_path = ""
-var marker_packs_array = Array()
+var root: TreeItem
+
+##########Node Connections###########
 onready var marker_packs = $Control/Dialogs/MarkerPacks/MarkerPacks
+onready var icons = $Icons
+onready var paths = $Paths
+onready var minimap = $Control/MiniMap
+
 
 func load_waypoint_markers(map_id):
-	self.marker_file_path = self.marker_file_dir + String(map_id) + ".data"
-	print("Loading protobuf file from path ", self.marker_file_path)
-	self.markerdata.clear_category()
-	self.markerdata.clear_icon()
-	self.markerdata.clear_trail()
+	self.marker_file_path = self.marker_file_dir.get_current_dir() + String(map_id) + ".data"
+	self.Waypoint_data.clear_category()
+	clear_map_markers()
+	root.free()
```

Comment:
You probably don't need to do this because in the next function `init_category_tree()` just overwrites it and the old data will be garbage collected.

Reply:
It turns out that without clearing it, its just continually adding to the tree. I moved `root.free()` into the function `init_category_tree()` if it is necessary part of the reseting the tree
![image](https://github.com/AsherGlick/Burrito/assets/89052698/f101da7a-032c-41a1-8c3b-07625385d170)

Reply:
this was never re-added anywhere